### PR TITLE
exclude Teleport license from the license check

### PR DESCRIPTION
**Pull request by hugoShaka**
This PR allows us to import teleport without causing a license error.

This is required for https://github.com/gravitational/teleport/pull/40224

### CODE REVIEW

File: .github/workflows/dependency-review.yaml
Comment (espadolini) [inline .github/workflows/dependency-review.yaml:61]: Would this work?
```suggestion
          allow-dependencies-licenses: pkg:golang/github.com/gravitational/teleport, pkg:golang/github.com/gravitational/teleport/api${{ inputs.allow-dependencies-licenses != '' && ', ${{inputs.allow-dependencies-licenses}}' }}
```

Comment (hugoShaka) [reply .github/workflows/dependency-review.yaml:61]: I'll let the folks with a PHD in GHA decide. I'm only copy-pasting things here 😅

**Review by jakule — APPROVED** [review]
I don't have a GH PhD but we're also blocked on this in TAG

**Review by r0mant — APPROVED** [review]

**Pull request merged**
